src/lib/HoneylemonService.js

```diff
@@ -0,0 +1,284 @@
+    ); // 10 days
+    const exchangeAddress = this.contractWrappers.exchange.address;
+
+    const order = {
+      makerAddress, // maker is the first address (miner)
+      takerAddress: NULL_ADDRESS, // taker is open and can be filled by anyone (when an investor comes along)
+      makerAssetAmount: sizeTh, // The maker asset amount
+      takerAssetAmount: sizeTh.multipliedBy(pricePerTh), // The taker asset amount
+      expirationTimeSeconds: new BigNumber(expirationTime), // Time when this order expires
+      makerFee: new BigNumber(0), // 0 maker fees
+      takerFee: new BigNumber(0), // 0 taker fees
+      feeRecipientAddress: NULL_ADDRESS, // No fee recipient
+      senderAddress: NULL_ADDRESS, // Sender address is open and can be submitted by anyone
+      salt: generatePseudoRandomSalt(), // Random value to provide uniqueness
+      makerAssetData: this.makerAssetData,
+      takerAssetData: this.takerAssetData,
+      exchangeAddress,
+      makerFeeAssetData: '0x',
+      takerFeeAssetData: '0x',
+      chainId: this.chainId
+    };
+
+    return order;
+  }
+
+  async signOrder(order) {
+    const signedOrder = await signatureUtils.ecSignOrderAsync(
+      this.provider,
+      order,
+      order.makerAddress
+    );
+
+    return signedOrder;
+  }
+
+  async submitOrder(signedOrder) {
+    return this.apiClient.submitOrderAsync(signedOrder);
+  }
+
+  async approveCollateralToken(makerAddress, amount) {
+    amount = amount || new BigNumber(2).pow(256).minus(1);
+    return this.collateralToken
+      .approve(this.minterBridgeAddress, amount)
+      .sendTransactionAsync({
+        from: makerAddress
+      });
+  }
+
+  async approvePaymentToken(takerAddress, amount) {
+    amount = amount || new BigNumber(2).pow(256).minus(1);
+    return this.paymentToken
+      .approve(this.contractWrappers.contractAddresses.erc20Proxy, amount)
+      .sendTransactionAsync({
+        from: takerAddress
+      });
+  }
+
+  async getCollateralTokenAmounts(makerAddress) {
+    const allowance = await this.collateralToken
+      .allowance(makerAddress, this.minterBridgeAddress)
+      .callAsync();
+    const balance = await this.collateralToken.balanceOf(makerAddress).callAsync();
+
+    return { allowance, balance };
+  }
+
+  async getPaymentTokenAmounts(takerAddress) {
+    const allowance = await this.paymentToken
+      .allowance(takerAddress, this.contractWrappers.contractAddresses.erc20Proxy)
+      .callAsync();
+    const balance = await this.paymentToken.balanceOf(takerAddress).callAsync();
+
+    return { allowance, balance };
+  }
+
+  async getOrderbook() {
+    const orderbookRequest = {
+      baseAssetData: this.makerAssetData,
+      quoteAssetData: this.takerAssetData
+    };
+    return this.apiClient.getOrderbookAsync(orderbookRequest);
+  }
+
+  async getOpenOrders(makerAddress) {
+    return this.apiClient.getOrdersAsync({ makerAddress });
+  }
+
+  async getOpenContracts(address) {
```

Comment:
Hey @chrismaree could you implement these methods? Not sure that signatures are correct, possibly we need to also pass a parameter indicating whether we're asking for long or short tokens. All other interactions should be ready - see `HoneylemonService.test.js`.

Reply:
I see, we don't have the taker amount in the bridge. I guess you can do it then. My concern was that we potentially can introduce a discrepancy between what actually happens and the events we report (say due to a bug). But it's probably ok to do it this way in the first version.